pkg/scheduler/framework/cycle_state.go

```diff
@@ -96,8 +106,9 @@ func (c *CycleState) Clone() *CycleState {
 }
 
 // Read retrieves data with the given "key" from CycleState. If the key is not
-// present an error is returned.
-// This function is thread safe by using sync.Map.
+// present, ErrNotFound is returned.
+//
+// See CycleState for notes on concurrency.
 func (c *CycleState) Read(key StateKey) (StateData, error) {
```

Comment:
in hindsight this may have been more idiomatic as a bool I think, but I suspect it is too late to correct it.

Reply:
Do you mean we should act like `func (c *CycleState) Read(key StateKey) (StateData, bool)`? But return an error is also a traditional pattern of golang.

Reply:
normal things that have a map back (like context.Context etc) don't return an error as the second result, rather a bool